parser/src/lib.rs

```diff
@@ -462,7 +462,7 @@ namespace N(2);
     }"#;
 
         let expected = r#"
-    impl<A, B> Iterator<ArrayIterator<A>, B> {
+    impl<A, B> Iterator<(ArrayIterator<A>), B> {
```

Comment:
Are these brackets necessary? The documentation of 
`format_type_args` says that they should only go in the last case if they are necessary (although that's not what the code actually does).

Reply:
Hm, interesting! I got a parse error, but only for a single parameter, let me check.